### PR TITLE
Fix #4928: BreadCrumb: accessibility

### LINKED ISSUE #4928: BreadCrumb: Improve accessibility
## 1) Environment
- PrimeFaces version: **latest**
- Does it work on the newest released PrimeFaces version? Version? **No**
- Does it work on the newest sources in GitHub? (Build by source -> https://github.com/primefaces/primefaces/wiki/Building-From-Source) **No**
- Application server + version: **-**
- Affected browsers: **-**

## 2) Expected behavior

According to the ARIA best practices site (https://www.w3.org/TR/wai-aria-practices/examples/breadcrumb/index.html) a breadcrumb should be:
- rendered as ordered list
- use no additional aria roles
- hide seperators
- set `aria-current="page"` for the last link

## 3) Actual behavior

The breadcrumbs are an unordered list using `menu` aria roles.

---
I fixed that for my project using a custom renderer and can provide a pull request if you think that this should be changed.

### PR DESCRIPTION
**Pull request by jepsar**
Fix #4928: BreadCrumb: accessibility

### CODE REVIEW

**Comment by melloware** [comment]
@jepsar Can you do me a favor and just test this Nova Light theme out locally by dropping this CSS into PF dir.

[nova-light.zip](https://github.com/primefaces/primefaces/files/5283598/nova-light.zip)


I want to see if I made it forward and backward compatible. I basically changed the SCSS from..
```css
.ui-breadcrumb {
        ul {
```

to

```css
.ui-breadcrumb {
        ul, ol {
```

**Review by melloware — APPROVED** [review]

**Comment by jepsar** [comment]
I'll have a look tomorrow.

The separator (chevron) also needs to be added in the style sheet.

**Comment by melloware** [comment]
OK here is the full SCSS for breacrumb...
```css
    .ui-breadcrumb {
        background-color: $panelContentBgColor;
        border: $panelContentBorder;
        padding: $panelContentPadding;
        
        ul, ol {
            li {
                float: none;
                display: inline-block;
                vertical-align: middle;
                
                .ui-menuitem-link {
                    color: $textColor;
                    margin: 0;
                    float: none;
                    padding: $menuitemPadding;

                    &:focus {
                        background-color: $menuitemHoverBgColor;
                        outline: 0 none;
                    }
                    
                    > span {
                        display: inline;
                        margin-right: $inlineSpacing / 2;
                    }
                }
                
                &.ui-breadcrumb-chevron {
                    @include icon_override("\e901");
                    font-size: $fontSize + 2;
                    margin-top: 2px;
                    color: $textSecondaryColor;
                }
                
                &:first-child {
                    a.ui-icon-home {
                        @include icon_override("\e925");
                        color: $textSecondaryColor;
                        margin: 0 $inlineSpacing 0 0;
                        font-size: $fontSize + 4;
                        vertical-align: top;
                        
                        span {
                            display: none;
                        }
                    }
                }
            }
        }
    }
```

**Comment by tandraschko** [comment]
We should talk with Primetek before

**Comment by melloware** [comment]
Agreed @mertsincan would have to make the theme changes but i just want @jepsar to verify my CSS changes are correct before I send them to him.  He would have to update the themes correctly.  But this change by default keeps things the way they are and you have to turn it on with accesible="true" so this way its backwards compatible.

**Comment by jepsar** [comment]
@melloware I don't think your CSS fixes the chevron as it is no longer an element in the HTML. I added some CSS to a draft PR https://github.com/primefaces/primefaces-showcase/pull/198

**Comment by tandraschko** [comment]
pinged mert last month about it, lets see

**Comment by tandraschko** [comment]
IMO we should not add a new attribute but always render the new markup if the theme was adjusted. Mert should decide here, i just dont think that 2 different markups are a good way to go.

**Comment by mertsincan** [comment]
Yes, I agree with Thomas. It can always be accessible. That's why we can use the ol tag. Also if you create a new Github issue with the "theme" label and 9.0 milestone I can add new styles to the designer API for breadcrumb. Thank you very much for this nice work.

Also, could @jepsar please remove "Note: currently this mode is not styled by default yet!" text? ;)

**Comment by jepsar** [comment]
@mertsincan I've removed the switch attribute. For a demo, see https://github.com/primefaces/primefaces-showcase/pull/198/files

Added new issues for styling: #6757

File: src/main/java/org/primefaces/component/breadcrumb/BreadCrumbRenderer.java [hunk 1]
```diff
@@ -28,6 +28,7 @@
 import javax.faces.component.UIComponent;
 import javax.faces.context.FacesContext;
 import javax.faces.context.ResponseWriter;
+import org.apache.commons.collections4.keyvalue.DefaultMapEntry;
```
Comment (melloware) [inline src/main/java/org/primefaces/component/breadcrumb/BreadCrumbRenderer.java]: We don't use Commons Collections right in PF? I don't think you can use this?

**Review by melloware — CHANGES_REQUESTED** [review]

File: src/main/java/org/primefaces/component/breadcrumb/BreadCrumbRenderer.java (hunk 1 above)
Comment (tandraschko) [reply src/main/java/org/primefaces/component/breadcrumb/BreadCrumbRenderer.java]: nope. how can it compile?

Comment (melloware) [reply src/main/java/org/primefaces/component/breadcrumb/BreadCrumbRenderer.java]: It must be part of a provided transitive dependency?

Comment (tandraschko) [reply src/main/java/org/primefaces/component/breadcrumb/BreadCrumbRenderer.java]: yeah from POI. can you please create a issue in general to hide this dependencies in PF?

Comment (melloware) [reply src/main/java/org/primefaces/component/breadcrumb/BreadCrumbRenderer.java]: Yep will create an issue.

Comment (jepsar) [reply src/main/java/org/primefaces/component/breadcrumb/BreadCrumbRenderer.java]: Sorry.. should have been `Map.entry`.. will fix this now

**Review by melloware — APPROVED** [review]

**Comment by mertsincan** [comment]
Hi @jepsar ,

Thanks a lot for your contribution ;)

**Pull request merged**